src/sagemaker/jumpstart/curated_hub/curated_hub.py

```diff
@@ -113,3 +155,193 @@ def delete_model(self, model_name: str, model_version: str = "*") -> None:
+                hub_model_version = Version(matched_model["version"])
+
+                # 1. Model version exists in Hub, pass
+                if hub_model_version == model_version:
+                    pass
+
+                # 2. Invalid model version exists in Hub, pass
+                # This will only happen if something goes wrong in our metadata
+                if hub_model_version > model_version:
+                    pass
+
+                # 3. Old model version exists in Hub, update
+                if hub_model_version < model_version:
+                    # Check minSDKVersion against current SDK version, emit log
+                    models_to_sync.append(model)
+
+        # Delete old models?
+
+        # Copy content workflow + `SageMaker:ImportHubContent` for each model-to-sync in parallel
+        tasks: List[futures.Future] = []
+        with futures.ThreadPoolExecutor(
+            max_workers=self._default_thread_pool_size,
+            thread_name_prefix="import-models-to-curated-hub",
+        ) as deploy_executor:
+            for model in models_to_sync:
+                task = deploy_executor.submit(self._sync_public_model_to_hub, model)
+                tasks.append(task)
+
+        # Handle failed imports
+        results = futures.wait(tasks)
+        failed_imports: List[Dict[str, Any]] = []
+        for result in results.done:
+            exception = result.exception()
+            if exception:
+                failed_imports.append(
+                    {
+                        "Exception": exception,
+                        "Traceback": "".join(
+                            traceback.TracebackException.from_exception(exception).format()
+                        ),
+                    }
+                )
+        if failed_imports:
+            raise RuntimeError(
+                f"Failures when importing models to curated hub in parallel: {failed_imports}"
+            )
+
+    def _sync_public_model_to_hub(self, model: Dict[str, str]):
+        """Syncs a public JumpStart model version to the Hub. Runs in parallel."""
+        model_name = model["model_id"]
+        model_version = model["version"]
+
+        model_specs = utils.verify_model_region_and_return_specs(
+            model_id=model_name,
+            version=model_version,
+            region=self.region,
+            scope=JumpStartScriptScope.INFERENCE,
+            sagemaker_session=self._sagemaker_session,
+        )
+        studio_specs = self._fetch_studio_specs(model_specs=model_specs)
+
+        dest_location = S3ObjectLocation(
+            bucket=self.hub_bucket_name, key=f"{model_name}/{model_version}"
+        )
+
+        src_files = ModelSpecsFileGenerator(self.region, self._s3_client, studio_specs).format(
+            model_specs
+        )
+        dest_files = S3PathFileGenerator(self.region, self._s3_client).format(dest_location)
+
+        sync_result = FileSync(src_files, dest_files, dest_location).call()
```

Comment:
Maybe rename to `sync_tasks`? since this is a list of tasks to complete by the handler.